idemp-base/src/org/bandahealth/idempiere/base/process/UpdateExpiredStockProcess.java

```diff
@@ -0,0 +1,57 @@
+package org.bandahealth.idempiere.base.process;
+
+import java.math.BigDecimal;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.logging.Level;
+
+import org.bandahealth.idempiere.base.model.MOrder_BH;
+import org.compiere.model.MAttributeSetInstance;
+import org.compiere.model.MStorageOnHand;
+import org.compiere.model.Query;
+import org.compiere.process.SvrProcess;
+
+/**
+ * This process resets expired stocks to zero for a given client
+ * 
+ * @author clinton
+ *
+ */
+
+public class UpdateExpiredStockProcess extends SvrProcess {
+
+    private final String PROCESS_NAME = this.getClass().getName();
+    
+    @Override
+    protected void prepare() {
+        
+    }
+
+    @Override
+    protected String doIt() throws Exception {
+        long start = System.currentTimeMillis();
+        log.log(Level.INFO, "START " + PROCESS_NAME);
+
+        int count = 0; 
+        String whereClause = MStorageOnHand.COLUMNNAME_QtyOnHand + "> 0";
+        List<MStorageOnHand> stocks = new Query(getCtx(), MStorageOnHand.Table_Name, whereClause, get_TrxName())
```

Comment:
You should only get stocks that are expired here.

Reply:
Done it in a different way since to know the stocks that are expired you have to check the date from Attribute Set Instance table